ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/behaviorTree/BehaviorTreeRootNodeExecutor.java

```diff
@@ -94,7 +95,22 @@ else if (state.getActionChildren().get(i).calculateExecuteAfterActionIndex() < e
       for (ActionNodeExecutor<?, ?> currentlyExecutingAction : currentlyExecutingActions)
       {
          currentlyExecutingAction.updateCurrentlyExecuting();
-         anyActionExecutionFailed |= currentlyExecutingAction.getState().getFailed();
+         boolean failed = currentlyExecutingAction.getState().getFailed();
+         anyActionExecutionFailed |= failed;
+
+         if (!currentlyExecutingAction.getState().getIsExecuting())
+         {
+            String name = currentlyExecutingAction.getDefinition().getName();
+            double elapsedExecutionTime = currentlyExecutingAction.getState().getElapsedExecutionTime();
+            if (failed)
+            {
+               LogTools.error("Action failed after %.2f s: %s".formatted(elapsedExecutionTime, name));
+            }
+            else
+            {
+               LogTools.info("Action completed successfully in %.2f s: %s".formatted(elapsedExecutionTime, name));
```

Comment:
There was no one place before where the event of success and failure were captured. This is important to begin implementing fallback behavior.